validator/validators/basic/basic_validator.go

```diff
@@ -0,0 +1,192 @@
+		metricSendingInterval = time.Minute
+		dataRate              = s.vConfig.GetDataRate()
+		agentCollectionPeriod = s.vConfig.GetAgentCollectionPeriod()
+		agentConfigFilePath   = s.vConfig.GetCloudWatchAgentConfigPath()
+		receivers             = s.vConfig.GetPluginsConfig()
+	)
+
+	if err := common.StartLogWrite(agentConfigFilePath, agentCollectionPeriod, metricSendingInterval, dataRate); err != nil {
+		multiErr = multierr.Append(multiErr, err)
+	}
+
+	// Sending metrics based on the receivers; however, for scraping plugin  (e.g prometheus), we would need to scrape it instead of sending
+	for _, receiver := range receivers {
+		if err := common.StartSendingMetrics(receiver, agentCollectionPeriod, metricSendingInterval, dataRate); err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	return multiErr
+}
+
+func (s *BasicValidator) CheckData(startTime, endTime time.Time) error {
+	var (
+		multiErr         error
+		ec2InstanceId    = awsservice.GetInstanceId()
+		metricNamespace  = s.vConfig.GetMetricNamespace()
+		validationMetric = s.vConfig.GetMetricValidation()
+		validationLog    = s.vConfig.GetLogValidation()
+	)
+
+	for _, metric := range validationMetric {
+		metricDimensions := []types.Dimension{
+			{
+				Name:  aws.String("InstanceId"),
+				Value: aws.String(ec2InstanceId),
+			},
+		}
+		for _, dimension := range metric.MetricDimension {
+			metricDimensions = append(metricDimensions, types.Dimension{
+				Name:  aws.String(dimension.Name),
+				Value: aws.String(dimension.Value),
+			})
+		}
+		err := s.ValidateMetric(metric.MetricName, metricNamespace, metricDimensions, metric.MetricValue, metric.MetricSampleCount, startTime, endTime)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	for _, log := range validationLog {
+		err := s.ValidateLogs(log.LogStream, log.LogValue, log.LogLines, startTime, endTime)
+		if err != nil {
+			multiErr = multierr.Append(multiErr, err)
+		}
+	}
+
+	return multiErr
+}
+
+func (s *BasicValidator) Cleanup() error {
+	var (
+		dataType      = s.vConfig.GetDataType()
+		ec2InstanceId = awsservice.GetInstanceId()
+	)
+	switch dataType {
+	case "logs":
+		awsservice.DeleteLogGroup(ec2InstanceId)
+	}
+
+	return nil
+}
+
+func (s *BasicValidator) ValidateLogs(logStream string, logLine string, numberOfLogLine int, startTime, endTime time.Time) error {
+	var (
+		logGroup = awsservice.GetInstanceId()
+	)
+	log.Printf("Start to validate log '%s' with number of logs lines %d within log group %s, log stream %s, start time %v and end time %v", logLine, numberOfLogLine, logGroup, logStream, startTime, endTime)
+	ok, err := awsservice.ValidateLogs(logGroup, logStream, &startTime, &endTime, func(logs []string) bool {
+		if len(logs) < 1 {
+			return false
+		}
+		actualNumberOfLogLines := 0
+		for _, l := range logs {
+			if strings.Contains(l, logLine) {
+				actualNumberOfLogLines += 1
+			}
+		}
+
+		return numberOfLogLine <= actualNumberOfLogLines
```

Comment:
Why is this <=?

Reply:
Yes [it has been discussed here](https://github.com/aws/amazon-cloudwatch-agent-test/pull/165#discussion_r1152239780).
```
It would be better to <= since there are logs we cannot control (e.g Security logs in Windows write every minute while System logs write every 5 minute). If in windows 2023 change the default interval, would == be true anymore? If we understand the changes sure; however, it would be better to <= to avoid flakiness and make it more deterministic
```